include/ignition/gui/qml/IgnSnackBar.qml

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.3
+import QtQuick.Window 2.2
+import QtQuick.Dialogs 1.0
+
+Popup {
+  id: snackbar
+  modal: duration == 0
+  focus: duration == 0
+  x: (window.width - width) / 2
+  y: window.height - window.height / 6
+  width: window.width - window.width / 6
+  contentHeight: notificationColumn.height
+
+  background: Rectangle {
+    border.color: "#444"
+    color: "#c9daf8"
+  }
+
+  // Duration of the snackbar. If duration is equal to zero then
+  // you should click somewhere in Ignition Gazebo to close it.
+  property int duration: 1000
```

Comment:
I'm finding 1 second too short to read some of the snackbars on this PR. The Material spec [recommends](https://material.io/components/snackbars#behavior) "_They automatically disappear from the screen after a minimum of four seconds, and a maximum of ten seconds._", so how about defaulting to `4000` and using that throughout this PR?